### PR TITLE
Updating NCC1701KB and adding via support

### PR DESCRIPTION
**Pull request by jessel92**
Bringing the NCC up to date with the latest QMK changes, as well as adding a keymap for Via support

## Description

Just cleaned up an old keyboard, that needed updating and also add a keymap for Via support

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by fauxpark — APPROVED** [review]

File: keyboards/themadnoodle/ncc1701kb/v2/keymaps/via/rules.mk
Comment (zvecr) [inline keyboards/themadnoodle/ncc1701kb/v2/keymaps/via/rules.mk:1]:
```suggestion
```

File: keyboards/themadnoodle/ncc1701kb/v2/keymaps/default/rules.mk
Comment (zvecr) [inline keyboards/themadnoodle/ncc1701kb/v2/keymaps/default/rules.mk:1]:
```suggestion
```

File: keyboards/themadnoodle/ncc1701kb/v2/info.json
Comment (zvecr) [inline keyboards/themadnoodle/ncc1701kb/v2/info.json:66]:
```suggestion
            { "pin_a": "D0", "pin_b": "D1" }
```

File: keyboards/themadnoodle/ncc1701kb/v2/info.json
Comment (zvecr) [inline keyboards/themadnoodle/ncc1701kb/v2/info.json:47]:
```suggestion
    "community_layouts": ["ortho_3x3"],
    "layouts": {
```

File: keyboards/themadnoodle/ncc1701kb/v2/config.h
Comment (zvecr) [inline keyboards/themadnoodle/ncc1701kb/v2/config.h:8]:
```suggestion
#define RGBLIGHT_DEFAULT_MODE RGBLIGHT_MODE_RAINBOW_SWIRL + 5
```

File: keyboards/themadnoodle/ncc1701kb/v2/rules.mk
Comment (zvecr) [inline keyboards/themadnoodle/ncc1701kb/v2/rules.mk:1]:
```suggestion
# This file intentionally left blank
```

**Review by zvecr — CHANGES_REQUESTED** [review]

File: keyboards/themadnoodle/ncc1701kb/v2/info.json
Comment (waffle87) [inline keyboards/themadnoodle/ncc1701kb/v2/info.json:65]:
```suggestion
```

File: keyboards/themadnoodle/ncc1701kb/v2/info.json
Comment (waffle87) [inline keyboards/themadnoodle/ncc1701kb/v2/info.json:15]:
```suggestion
        "rgblight": true,
        "encoder": true
```

**Review by jessel92 — COMMENTED** [review]
Everything should be good to go now! Let me know if this is all correct!

File: keyboards/themadnoodle/ncc1701kb/v2/readme.md
Comment (sigprof) [inline keyboards/themadnoodle/ncc1701kb/v2/readme.md:3]:
```suggestion
![NCC1701KBv2](https://i.imgur.com/Zq1SjBTh.jpg)
```
A link to an Imgur album does not work here — you need a direct link to a single image (with `h` added at the end of the image ID to limit the image size).

It is possible to wrap the image link into a hyperlink to the whole album, but in this case the album contains only a single image, so the album link is not very useful.

**Review by sigprof — CHANGES_REQUESTED** [review]

File: keyboards/themadnoodle/ncc1701kb/v2/readme.md
Comment (jessel92) [reply keyboards/themadnoodle/ncc1701kb/v2/readme.md]: Ah, ok, I didnt realize that. Thanks for the heads up!

**Review by waffle87 — APPROVED** [review]

**Review by fauxpark — CHANGES_REQUESTED** [review]
Keyboard updates need to go through `develop`.

**Comment by jessel92** [comment]
> Keyboard updates need to go through `develop`.

@fauxpark changed the base branch to develop. I didn't think this needed to go through the develop branch since no files or directories changed names. Let me know if this works. I'm trying to get these through ASAP.

**Comment by fauxpark** [comment]
It's not just about file and folder names - since `master` gets merged into `develop` automatically there is a potential for merge conflicts if the same files are modified in `develop`. So really only new keyboard additions and urgent bugfixes can go to master.

**Comment by jessel92** [comment]
> It's not just about file and folder names - since `master` gets merged into `develop` automatically, there is a potential for merge conflicts if the same files are modified in `develop`. So really only new keyboard additions and urgent bugfixes can go to master.
> 
> You have some merge conflicts due to the merge commit you just made. This is not necessary because of the above. Please remove it.

Oh, ok I understand! But the merge conflicts are another story, should I switch back to the master branch? Because it says only those with write access can "resolve conflicts"? what should I do to fix this?

**Comment by jessel92** [comment]
> It's not just about file and folder names - since `master` gets merged into `develop` automatically there is a potential for merge conflicts if the same files are modified in `develop`. So really only new keyboard additions and urgent bugfixes can go to master.
> 
> You have some merge conflicts due to the merge commit you just made. This is not necessary because of the above. Please remove it.

@fauxpark Oh, ok I understand! But the merge conflicts are another story, should I switch back to the master branch? Because it says only those with write access can "resolve conflicts"? what should I do to fix this?

**Comment by fauxpark** [comment]
You just need to undo the last commit.

**Comment by jessel92** [comment]
@fauxpark  the last commit was a readme.md commit. or are you talking about switching branches?

**Comment by fauxpark** [comment]
image

**Comment by jessel92** [comment]
> image

@fauxpark ok o reverted it, but it now looks like i have a "python" tag also this a ton of conflcs

**Comment by jessel92** [comment]
@fauxpark it seems like changing the base branch of the PR REALLY messed things up.

**Comment by fauxpark** [comment]
No, there must be something wrong with your branch. Although it doesn't seem to have diverged, it's only 14 commits ahead of master.

It might be better to just resubmit this PR pointing at develop.

**Comment by jessel92** [comment]
@fauxpark Yes, it was totally fine before the PRs base branch change. If I open a new a PR is there any way that we can push it through so I don't have to go through all that again?

**Comment by jessel92** [comment]
@fauxpark Even when i try to create a new branch on my fork and make a new PR to the develop branch, I get HUNDREDs of commits that are not mine. What's going on? Is this an issue on my end or yours? Im just trying to clean up my keymaps and make an old keypad fit the new QMK json style.

**Comment by fauxpark** [comment]
I'm going to close this PR. Please resubmit it properly.

**Comment by fauxpark** [comment]
Looks like the master->develop merged failed due to conflicts (related to the ongoing effort to remove user keymaps, apparently those PRs are going to master for some reason). So that explains what all the mess here is about.

**Comment by jessel92** [comment]
> Looks like the master->develop merged failed due to conflicts (related to the ongoing effort to remove user keymaps, apparently those PRs are going to master for some reason). So that explains what all the mess here is about. I've fixed it up; develop should now contain all of master's history.

@fauxpark Ugh that's rough, but thank you so much for helping through all the hassle! The new PR was a success!

**Comment by jessel92** [comment]
> Looks like the master->develop merged failed due to conflicts (related to the ongoing effort to remove user keymaps, apparently those PRs are going to master for some reason). So that explains what all the mess here is about. I've fixed it up; develop should now contain all of master's history.

@fauxpark Ugh that's rough, but thank you so much for helping through all the hassle! The new PR was a success! I have a few other PRs open right now that I think you're reviewing as well. I hope they don't cause as much trouble and we can get them done quickly! I think you're up for review on Noodlepad Additions and Updates #22701 ;-)

**Pull request closed without merging**
